mods/nyancat/init.lua

```diff
@@ -0,0 +1,93 @@
+minetest.register_node("nyancat:nyancat", {
+	description = "Nyan Cat",
+	tiles = {"default_nc_side.png", "default_nc_side.png", "default_nc_side.png",
+		"default_nc_side.png", "default_nc_back.png", "default_nc_front.png"},
+	paramtype2 = "facedir",
+	groups = {cracky = 2},
+	is_ground_content = false,
+	legacy_facedir_simple = true,
+	sounds = default.node_sound_defaults(),
+})
+
+minetest.register_node("nyancat:nyancat_rainbow", {
+	description = "Nyan Cat Rainbow",
+	tiles = {
+		"default_nc_rb.png^[transformR90", "default_nc_rb.png^[transformR90",
+		"default_nc_rb.png", "default_nc_rb.png"
+	},
+	paramtype2 = "facedir",
+	groups = {cracky = 2},
+	is_ground_content = false,
+	sounds = default.node_sound_defaults(),
+})
+
+minetest.register_craft({
+	type = "fuel",
+	recipe = "nyancat:nyancat",
+	burntime = 1,
+})
+
+minetest.register_craft({
+	type = "fuel",
+	recipe = "nyancat:nyancat_rainbow",
+	burntime = 1,
+})
+
+nyancat = {}
+
+function nyancat.place(pos, facedir, length)
+	local tailvec = {x = 0, y = 0, z = 0}
+	if facedir == 0 then
```

Comment:
do you mean 

```lua
local tailvec = minetest.facedir_to_dir(facedir)
```

then remove the long if chain?

That's a fairly unrelated change